Add Dashboard Widgets

circlecube opened the pull request:
## Proposed changes

This adds new dashboard widgets for Bluehost Account links and Bluehost Help. These are rebuilt from the Bluehost App in PHP since there is no need to create a react app to render them here.

The widgets each enqueue the app stylesheet to reuse the nfd tailwind classes. There is some custom CSS in the view file, but I tried to use any existing classes when possible.

The account widget lists links to bluehost account - these links are dependant on jarvis and bluehost India, but I'm not sure the state of these links. I believe we can simplify the logic, but am not current on them. I plan to follow up with some simplified logic (what I have here is copied from the logic in the current js app. The links open in new _blank windows, have UTM params, are jarvis aware and brand aware (for bluehost India sites).

The help widget checks if site is HelpCenter-capable, and opens the help center when possible. Currently if HelpCenter is not supported, this would link to the help page, which will probably be discontinued so should we have a fallback here? Maybe a help link in their bluehost account?

I set it up so we could easily add more widgets to the widgets/bootstrap.php file by adding the class and the view. The other widgets will likely need a js app build since they will be interactive, but that somewhat remains TBD. 

Here's a screenshot of how they look:
![bluehost-widgets](https://github.com/user-attachments/assets/d3f67f05-5086-4514-b47c-978e5670324e)

vs the design file:
![Captura de pantalla 2025-02-05 a las 12 49 59](https://github.com/user-attachments/assets/5055f545-f3da-4a37-aed2-ad0e95b8974e)

Assuming we can ship this once it is ready and keep the current app components in place. But can also update this PR to simultaneously pull out those components.

TODO: Still need to add cypress tests to check on these widgets.

## Type of Change

<!-- What types of changes does your code introduce? -->
<!-- _Put an `x` in the boxes that apply_ -->

#### Production

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update
- [ ] Refactoring / housekeeping (changes to files not directly related to functionality)

<!-- Bugfixes should explain how to reproduce the bug -->

#### Development

- [ ] Tests
- [ ] Dependency update
- [ ] Environment update / refactoring
- [ ] Documentation Update

<!-- All PRs should endeavor to include tests -->
<!-- PRs with new tools or dependencies should explain how to use them -->

## Video

<!-- Add a short video demonstrating where the change takes effect and how to can be reproduced by reviewers -->
<!-- On macOS press cmd+shift+5 to open the screen recording tool. It will save the video to the desktop  -->

<!-- Drag and drop the video file into the GitHub editor to attach it -->

## Checklist

<!-- _Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._ -->

- [ ] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [ ] I have viewed my change in a web-browser
- [ ] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

<!-- 
## Pre-deployment Checklist

- [ ] E.g. environmental variables which need to be set before deployment will work.
- [ ] E.g. other tickets which need to be complete before an API this change relies on will be ready
-->

<!--
## Post-deployment Checklist

How can the change be verified?

- [ ] E.g. temporarily enable logging and observe specific logs.
- [ ] E.g. observe new data in a specific view or table.
-->

## Further comments

<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->
<!-- Now is a good time to create additional tickets for any necessary follow-up work. -->

wpscholar commented on inc/widgets/views/account.php:
```diff
@@ -0,0 +1,246 @@
+ * @param string $url the URL
+ * @return string the URL with UTM params
+ */
+function add_utm_params( $url ) {
```
@circlecube We should move all functions to static methods on the widget implementation class.

circlecube replied:
Good call, got them moved to the main class in each case.

wpscholar commented:
@circlecube We should clean up the linting issues. Also, we should add a few test ID attributes (e.g. `data-cy`, `data-test`, `data-testid`). We should also decide on the best naming. I'm partial to `data-cy`.

circlecube commented:
I added `data-cy` attributes to the buttons.

Also, added the site preview widget to this PR. Here's previews when coming soon is enabled and disabled:
![Screenshot 2025-02-18 at 5 55 40 PM](https://github.com/user-attachments/assets/235629b9-c25f-4b62-99c7-0d59ca625e3a)
![Screenshot 2025-02-18 at 5 55 54 PM](https://github.com/user-attachments/assets/232c890a-4b8b-4d1d-b5ff-10e5e55d2718)
